src/test/rustdoc-json/nested.rs

```diff
@@ -1,7 +1,25 @@
 // edition:2018
+
+// @has nested.json "$.index.['0:0'].kind" \"module\"
+// @has - "$.index.['0:0'].inner.is_crate" true
+// @has - "$.index.['0:0'].inner.items[*]" \"0:3\"
+
+// @has nested.json "$.index.['0:3'].kind" \"module\"
+// @has - "$.index.['0:3'].inner.is_crate" false
+// @has - "$.index.['0:3'].inner.items[*]" \"0:4\"
+// @has - "$.index.['0:3'].inner.items[*]" \"0:7\"
```

Comment:
This makes me very nervous. If the compiler changes how DefIds are generated, this will break.

Can you instead find `l1` by name, and just assert that the ID exists, not the format?

Reply:
Do you mean not using the `.index.['n:m']` part, or the `items[*] "n:m"` part? The first one I could do by using filter expressions, though it would make the paths a little more verbose. The second I actually think it might be better to use `@count` and just say 'there are two things in the items list', as I can't think of a good way to assert they're the right items aside from hardcoding the IDs

Reply:
I mean hard-coding the DefIds anywhere, so both. `@count` sounds good to me.